app/src/main/java/org/cru/godtools/fragment/ToolDetailsFragment.java

```diff
@@ -306,6 +329,24 @@ private void updateViews() {
         }
     }
 
+    private void updateOverviewVideo() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+            if (mVideoBanner != null) {
+                mVideoBanner.setEnableAutomaticInitialization(false);
+                getLifecycle().addObserver(mVideoBanner);
+                mVideoBanner.addYouTubePlayerListener(new AbstractYouTubePlayerListener() {
+                    @Override
+                    public void onReady(@NotNull final YouTubePlayer youTubePlayer) {
+                        if (mTool != null && mTool.getOverviewVideo() != null) {
+                            String videoId = mTool.getOverviewVideo().split("=")[1];
```

Comment:
this is a really brittle way to handle this

Reply:
we also need to do something to try and allow this to be forward compatible if we change where we host these overview videos in the future.

Reply:
there are 2 issues here.
1. youtube video urls come in many forms, splitting on `=` to get the video id is incredibly brittle. (e.g. these video link will fail: https://youtu.be/dQw4w9WgXcQ https://www.youtube.com/watch?v=dQw4w9WgXcQ&feature=youtu.be)
2. how do we handle changing from youtube to another provider in the future without breaking existing apps.

issue 2 can be addressed by tweaking the attribute name in the API and you would just have to update the JsonApi annotation. Issue 1 needs a bit more robust youtube link parsing, along with treating an unhandled link format as having no overview video.

Reply:
I added youtube urls to 3 more tools on stage, 2 of them are valid but different forms, and 1 is not a video link and should be invalid.

Reply:
I updated the attribute name to be `attr-about-overview-video-youtube`

Reply:
@frett I think it would be best just to place the youtube VideoId in the API.  Since we can easily recreate a URL if we need to also this will eliminate the first issue.  To resolve issue number two we can set the JSON attribute to `attr-about-overview-video-youtube`. This way we can change the trailing text to match the provider.

Reply:
I was thinking this as well and I'm a bit torn. we can discuss briefly this afternoon

Reply:
talked with Ann, she is fine with just using the youtube video id, so I updated the attribute to return the video id only